core/src/main/java/hudson/scheduler/BaseParser.java

```diff
@@ -70,15 +70,23 @@ public void setHash(Hash hash) {
     protected long doRange(int start, int end, int step, int field) throws ANTLRException {
         rangeCheck(start, field);
         rangeCheck(end, field);
+        if (field == 1 && start>end) // Hour-value passes midnight
+            end += 24;
```

Comment:
Does this mean `0 21-27 * * *` is also legal now?

Reply:
No, why should you think that? The call to 'rangeCheck' just above still makes sure both numbers are between there respective minimum and maximum value (0 and 23).